### ISSUE TITLE
`_RP` directory in reproducible build script should not be required

### ISSUE THREAD

**Issue by sxa** [adoptium/temurin-build#4014]
**What are you trying to do?** Run the reproducible comparison tooling in `tooling/reproducible`

**Expected behaviour:** Temporary directory `_RP` under the current working directory should not be left around. It's not completely clear why this is required separate from the `_CP `ones. 

**Observed behaviour:** `_BK` location under the current directory is created and left around after the build. Also the login around it in https://github.com/adoptium/temurin-build/blob/62ecfbaeb60cbe5f4072bd81b8d4a3546625dcb2/tooling/reproducible/repro_compare.sh#L24 seems incorrect - `JDK_DIR` is not necessarily defined which causes the directory to be created as just `_BK` under the current location.

**Any other comments:** Related: https://github.com/adoptium/temurin-build/issues/4013

**Comment by sxa** [comment]
@sophia-guo Can you take a look at this so I can understand the impact if this `_BK` and the `JDK_DIR` references are removed? There is no `JDK_DIR` directory defined at this point so the script leaves you with `_BK` directories on the file system. It seems to have been introduced into `repro_compare.sh` in commit https://github.com/adoptium/temurin-build/commit/08621c72262ba260e4bb6451a9c75bf1c0ab365d#diff-baa6ea4d277f52c24e6205a75e767f1dae477d0054d8fba3550e3551a4ae3eb3R54 as part of https://github.com/adoptium/temurin-build/pull/3972

It looks like the related `BK_JDK_DIR` is only used in:
```
    removeSystemModulesHashBuilderParams "${JDK_DIR}" "${OS}" "${BK_JDK_DIR}"
    processModuleInfo "${JDK_DIR}" "${OS}" "${BK_JDK_DIR}"
```
Is the intended purpose of `_BK` different from `_CP` directory which is created elsewhere and can we re-use that instead?

**Comment by sophia-guo** [comment]
No, _BK and _CP are created for same purpose to provide workable java or other jdk tools. Keep the  _BK ( BK_JDK_DIR should be ${JDK_DIR1}_BK)  one should be good I think.
